Audit preparation (#121)
NestedFi/nested-core-lego: contracts/libraries/CURVE_POOLS.md

- [Curve factory pools](https://curve.readthedocs.io/factory-pools.html) - pools pool with WETH
- [Curve factory pools](https://curve.readthedocs.io/factory-pools.html) - pools pool without ETH or WETH

## Differences betweens Curve pools

### int128 & uint256
Some Curve pools has a different interfaces especially on the `remove_liquidity_one_coin` function.
You can find those two different function selectors:

```
remove_liquidity_one_coin(uint256,int128,uint256)
remove_liquidity_one_coin(uint256,uint256,uint256)
```

These differences are the consequence of the presence of functions `withdraw128()` and `withdraw256()` in the operator.

### ETH managment

#### Add liquidity

Some Curve pools has ETH as liquidity, and require to call `add_liquidity` with a value to use ETH for the liquidity addition.
So to add liquidity to this type of pool using ETH, we must use `depositETH(address,uint256,uint256)` that will call:

```
add_liquidity{value: x}(uint256[],uint256)
```

And to add liquidity in this type of pool using another token, we must call:

```
add_liquidity(uint256[],uint256)
```

#### remove liquidity

> To withdraw liquidity and receive ETH, we have to withdraw some ETH from WETH contract because the [ETH/WETH nested protocol managment](https://github.com/NestedFi/nested-core-lego#eth-managment) forces operators to handle WETH as input, so we must call:

Author: ```suggestion
When you remove liquidity from a Curve pool using ETH as a Curve output token, the `NestedFactory` will automaticly convert the received ETH into WETH because the [ETH/WETH nested protocol managment](https://github.com/NestedFi/nested-core-lego#eth-managment) forces WETH conversion on `receive` if the sender is not the whithdrawer, so when you call `withdrawETH(address,uint256,uint256)`, you will receive WETH as output token even if you asked for ETH.
```

> ```

Author: ```suggestion
```

> withdrawETH(address,uint256,uint256)

Author: ```suggestion
```

> ```

Author: ```suggestion
```
